l3_dump.py

```diff
@@ -227,8 +226,26 @@ def main():
     """
     Shell to call do_main() with command-line arguments.
     """
-    check_usage()
-    (program_bin, decode_loc_id, loc_decoder_bin) = set_decode_loc_id()
+    do_main(sys.argv[1:])
+
+###############################################################################
+def do_main(args:list, return_logentry_lists:bool = False) -> int:
+    """
+    Main method that drives the unpacking of the L3-dump file to print
+    diagnostic data. This modularized method exists outside of main() so that
+    it can be called independently via pytests.
+
+    Arguments:
+        args[0] - Str: Name of L3-log file
+        args[1] - Str: Name of binary that generated L3-log file.
+
+    Returns: Tuple:
+        # entries processed
```

Comment:
We should probably document the return tuple.

Reply:
Done. Corrected the return type signature, as well.